### PR TITLE
Fix Windows test suites Network.Socket and Network.Inet.TCP test case…

### PR DESCRIPTION
**Pull request by rnjtranjan**
…s are hanging for ever

### CODE REVIEW

**Comment by rnjtranjan** [comment]
Fixes:https://github.com/composewell/streamly/issues/2315

File: test/Streamly/Test/Network/Inet/TCP.hs [hunk 1]
```diff
@@ -135,10 +135,15 @@ moduleName :: String
...
+main =
+#if !defined(mingw32_HOST_OS)
+    hspec $ do
     modifyMaxSuccess (const 1) $ do
       describe moduleName $ do
         describe "Accept Connections" $ do
             prop "acceptOnPort" validateOnPort
             prop "acceptOnPortLocal" validateOnPortLocal
         --  prop "acceptOnAddr/connect" Tested as part of above test cases
+#else
```
Comment (harendra-kumar) [inline test/Streamly/Test/Network/Inet/TCP.hs:147]: I thought there are only 1 or 2 tests failing, and there are many others that are not failing.

Since all tests are failing, it makes sense to disable `-threaded` for windows case so that we at least test something.

File: test/streamly-tests.cabal [hunk 2]
```diff
@@ -364,28 +364,29 @@ test-suite FileSystem.Handle
...
+  import: lib-options
   type: exitcode-stdio-1.0
   main-is: Streamly/Test/Network/Inet/TCP.hs
-  if flag(use-streamly-core)
-    buildable: False
-  -- XXX on Windows these test cases are hanging for ever
-  -- need to be investigated.
-  -- https://github.com/composewell/streamly/issues/2315
   if os(windows)
+    ghc-options:-rtsopts -fno-ignore-asserts
```
Comment (harendra-kumar) [inline test/streamly-tests.cabal:371]: This can be taken out of the condition, it is common to both.

File: test/streamly-tests.cabal [hunk 3]
```diff
@@ -364,28 +364,29 @@ test-suite FileSystem.Handle
...
-  -- need to be investigated.
-  -- https://github.com/composewell/streamly/issues/2315
   if os(windows)
-    buildable: False
+    ghc-options:-rtsopts -fno-ignore-asserts
+  else
+    ghc-options:-rtsopts -fno-ignore-asserts -threaded -with-rtsopts=-N
+  include-dirs: .
+  build-depends: streamly-tests
+  if flag(use-streamly-core)
```
Comment (harendra-kumar) [inline test/streamly-tests.cabal:389]: There is nothing in this condition?

File: test/streamly-tests.cabal [hunk 4]
```diff
@@ -364,28 +364,29 @@ test-suite FileSystem.Handle
...
-  if flag(use-streamly-core)
-    buildable: False
-  -- XXX on Windows these test cases are hanging for ever
-  -- need to be investigated.
-  -- https://github.com/composewell/streamly/issues/2315
   if os(windows)
+    ghc-options:-rtsopts -fno-ignore-asserts
+  else
+    ghc-options:-rtsopts -fno-ignore-asserts -threaded -with-rtsopts=-N
+  include-dirs: .
```
Comment (harendra-kumar) [inline test/streamly-tests.cabal:374]: Why is this required now, and was not required earlier?

Comment (rnjtranjan) [reply test/streamly-tests.cabal:371]: It came from test-options.
